src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -4,30 +4,28 @@
  */
 
 import * as vscode from 'vscode'
-import * as codewhispererClient from '../../client/codewhisperer'
 import * as fs from 'fs-extra'
 import { DependencyGraph } from '../dependencyGraph/dependencyGraph'
-import { BMDocument, performBM25Scoring } from './rankBm25'
-import { getRelevantFilesFromEditor, isRelevant } from './editorFilesUtil'
+import { BM25Document, BM25Okapi } from './rankBm25'
+import { isRelevant } from './editorFilesUtil'
 import { ToolkitError } from '../../../shared/errors'
-import { supplemetalContextFetchingTimeoutMsg } from '../../models/constants'
+import { crossFileContextConfig, supplemetalContextFetchingTimeoutMsg } from '../../models/constants'
 import { CancellationError } from '../../../shared/utilities/timeoutUtils'
+import { CodeWhispererSupplementalContextItem } from './supplementalContextUtil'
 
 const crossFileLanguageConfigs = ['java']
 interface Chunk {
     fileName: string
     content: string
     nextContent: string
+    score?: number
 }
-const chunkSize = 10
-const chunkCount = 60
-const topK = 3
 
 export async function fetchSupplementalContextForSrc(
```

Comment:
Do you have unit tests for `fetchSupplementalContextForSrc`?

Reply:
not for now

Reply:
In fact, I think it's even better to refactor the overall structure of the cross file context / utg feature in vscode, I would prefer more abstraction instead of exporting functions everywhere.

Reply:
I will take it as a follow up once I have bandwidth to do it.